examples/some-verifier/src/main.rs

```diff
@@ -315,9 +256,103 @@ async fn add_verification(
+    let num_statements = proof.verifiable_credential.len();
+    if num_statements < 2 {
+        return Err(Error::NotEnoughStatements(num_statements));
+    }
+
+    let mut entry = VerificationsEntry::from_presentation(&proof);
+    for proof in &proof.verifiable_credential {
+        proof_to_verifications_entry(state.clone(), proof, &mut entry)?;
+    }
+
+    match state.database.add_verification(entry).await {
+        Ok(()) => {}
+        // TODO: duplicate/overlapping entries
+        Err(err) => {
+            tracing::warn!("Error inserting entries: {err}");
+            return Err(Error::DatabaseError(err));
+        }
+    }
+
+    tracing::info!("Successfully verified.");
+
+    Ok(StatusCode::CREATED)
+}
+
+#[tracing::instrument(level = "info", skip_all)]
+async fn remove_verification(
+    State(mut state): State<AppState>,
+    request: Result<Json<Request>, JsonRejection>,
+) -> Result<StatusCode, Error> {
+    tracing::info!("Request to remove verification.");
+
+    let request = request?;
+    verify_request(&mut state, &request).await?;
+
+    let Json(Request { proof, .. }) = request;
+
+    // Check the statements and add them to the database
+    let num_creds = proof.verifiable_credential.len();
+    if num_creds != 1 {
+        return Err(Error::NotSingleStatement(num_creds));
+    }
+
+    let credential = proof.verifiable_credential.get(0).unwrap(); // We know we have exactly 1 credential
+    let (id, platform) = match credential {
+        CredentialProof::Web3Id {
+            network,
+            contract,
+            proofs,
+            ..
+        } => {
+            if network != &state.network {
+                return Err(Error::InvalidIssuer);
+            }
+            let platform = match contract {
+                addr if addr == &state.telegram_registry => Platform::Telegram,
+                addr if addr == &state.discord_registry => Platform::Discord,
+                _ => return Err(Error::InvalidIssuer),
+            };
+
+            let id = match &proofs[..] {
+                [(
+                    _,
+                    AtomicProof::RevealAttribute {
+                        attribute: Web3IdAttribute::String(AttributeKind(id)),
+                        ..
+                    },
+                )] => id.clone(),
+                _ => {
+                    return Err(Error::InvalidStatement);
+                }
+            };
+
+            (id, platform)
+        }
+        _ => return Err(Error::InvalidStatement),
+    };
+
+    if let Err(err) = state.database.remove_verirication(&id, platform).await {
+        tracing::warn!("Error inserting entries: {err}");
+        return Err(Error::DatabaseError(err));
+    }
+
+    tracing::info!("Successfully removed verification.");
+
+    Ok(StatusCode::OK)
+}
 
-    let delta = Utc::now().signed_duration_since(timestamp);
+async fn verify_request(state: &mut AppState, request: &Json<Request>) -> Result<(), Error> {
```

Comment:
There is no reason to take a `&Json<Request>` here. Changing it to `&Request` will also help simplify the code at the call sites.